sympy/polys/multivariate_resultants.py

```diff
@@ -126,13 +121,17 @@ def get_dixon_polynomial(self):
 
         return poly_from_expr(dixon_polynomial, self.dummy_variables)[0]
 
-    def get_upper_degree(self):
-        list_of_products = [self.variables[i] ** self.max_degrees[i]
-                            for i in range(self.n)]
-        product = prod(list_of_products)
-        product = Poly(product).monoms()
+    def get_max_degrees(self, polynomial):
+        r"""
+        Returns a list of the maximum degree of each variable appearing
+        in the coefficients of the Dixon polynomial. The coefficients are
+        viewed as polys in x_1, ... , x_n.
+        """
+        max_degrees = [
+            max(degree_list(Poly(poly, self.variables))[i] for poly
+                in polynomial.coeffs()) for i in range(self.n)]
```

Comment:
It seems that, for each `i`, a new `Poly(poly, self.variables)` objects will be created for each `poly`, which is inefficient. I would first create all degree lists

    deg_lists = [degree_list(Poly(poly, self.variables))
                 for poly in polynomial.coeffs()]

and then compute the maxima for each `i`. That should be possible with something like this:

    max_degrees = [max(degs) for degs in zip(*deg_lists)]

(You can think of `deg_lists` as a matrix where the maximum of each column should be computed. `zip` will essentially transpose the matrix and turn the columns into rows `degs`.)

Reply:
Oh, I see. Thanks for the review. I really liked your explanation for the code above. I'll fix it right away.